depends/packages/boost.mk

```diff
@@ -5,6 +5,102 @@ $(package)_file_name=boost_$(subst .,_,$($(package)_version)).tar.bz2
+  cp boost/noncopyable.hpp $($(package)_staging_prefix_dir)/include/boost && \
+  cp boost/none.hpp $($(package)_staging_prefix_dir)/include/boost && \
+  cp boost/none_t.hpp $($(package)_staging_prefix_dir)/include/boost && \
+  cp -r boost/numeric $($(package)_staging_prefix_dir)/include/boost && \
+  cp boost/operators.hpp $($(package)_staging_prefix_dir)/include/boost && \
+  cp boost/optional.hpp $($(package)_staging_prefix_dir)/include/boost && \
+  cp -r boost/optional $($(package)_staging_prefix_dir)/include/boost && \
+  cp boost/predef.h $($(package)_staging_prefix_dir)/include/boost && \
+  cp -r boost/predef $($(package)_staging_prefix_dir)/include/boost && \
+  cp -r boost/preprocessor $($(package)_staging_prefix_dir)/include/boost && \
+  cp boost/process.hpp $($(package)_staging_prefix_dir)/include/boost && \
+  cp -r boost/process $($(package)_staging_prefix_dir)/include/boost && \
+  cp boost/scoped_array.hpp $($(package)_staging_prefix_dir)/include/boost && \
+  cp boost/scoped_ptr.hpp $($(package)_staging_prefix_dir)/include/boost && \
+  cp -r boost/smart_ptr $($(package)_staging_prefix_dir)/include/boost && \
+  cp boost/swap.hpp $($(package)_staging_prefix_dir)/include/boost && \
+  cp -r boost/range $($(package)_staging_prefix_dir)/include/boost && \
+  cp boost/ref.hpp $($(package)_staging_prefix_dir)/include/boost && \
+  cp boost/shared_ptr.hpp $($(package)_staging_prefix_dir)/include/boost && \
+  cp -r boost/signals2 $($(package)_staging_prefix_dir)/include/boost && \
+  cp -r boost/system $($(package)_staging_prefix_dir)/include/boost && \
+  cp boost/static_assert.hpp $($(package)_staging_prefix_dir)/include/boost && \
+  cp boost/tokenizer.hpp $($(package)_staging_prefix_dir)/include/boost && \
+  cp boost/token_functions.hpp $($(package)_staging_prefix_dir)/include/boost && \
+  cp boost/token_iterator.hpp $($(package)_staging_prefix_dir)/include/boost && \
+  cp -r boost/tuple $($(package)_staging_prefix_dir)/include/boost && \
+  cp boost/type.hpp $($(package)_staging_prefix_dir)/include/boost && \
+  cp boost/type_index.hpp $($(package)_staging_prefix_dir)/include/boost && \
+  cp -r boost/type_index $($(package)_staging_prefix_dir)/include/boost && \
+  cp -r boost/type_traits $($(package)_staging_prefix_dir)/include/boost && \
+  cp -r boost/test $($(package)_staging_prefix_dir)/include/boost && \
+  cp boost/throw_exception.hpp $($(package)_staging_prefix_dir)/include/boost && \
+  cp -r boost/utility $($(package)_staging_prefix_dir)/include/boost && \
+  cp boost/utility.hpp $($(package)_staging_prefix_dir)/include/boost && \
+  cp -r boost/variant $($(package)_staging_prefix_dir)/include/boost && \
+  cp boost/version.hpp $($(package)_staging_prefix_dir)/include/boost && \
+  cp boost/visit_each.hpp $($(package)_staging_prefix_dir)/include/boost && \
+  cp boost/weak_ptr.hpp $($(package)_staging_prefix_dir)/include/boost && \
+  cp -r boost/winapi $($(package)_staging_prefix_dir)/include/boost
```

Comment:
How painful is it to update this when boost changes?

Reply:
It's not too awful, and should mostly be removing headers (there are already a few we can drop in the next Boost version). Note that the removal of the winapi dir here is because boost ships it twice (one in boost/winapi and another copy in boost/detail/winapi). If we can (hopefully) get rid of external signer there are a large amount of headers we could drop from here.